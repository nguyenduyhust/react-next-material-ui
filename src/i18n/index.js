const NextI18Next = require('next-i18next').default;
const path = require('path');

module.exports = new NextI18Next({
  defaultLanguage: 'en',
  otherLanguages: ['ja'],
  browserLanguageDetection: true,
  serverLanguageDetection: true,
  detection: {
    order: ['querystring', 'cookie', 'header', 'navigator'],
    lookupQuerystring: 'lng',
    lookupCookie: 'lng',
    lookupHeader: 'accept-language',
    caches: ['cookie'],
  },
  localePath: path.resolve('public/static/locales'),
});
