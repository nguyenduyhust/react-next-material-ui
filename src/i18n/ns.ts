export enum NamespaceEnum {
  COMMON = 'common',
  HOME_PAGE = 'homepage',
}
export const ns: string | Array<string> = [NamespaceEnum.COMMON, NamespaceEnum.HOME_PAGE];
export const defaultNS: string = NamespaceEnum.COMMON;
