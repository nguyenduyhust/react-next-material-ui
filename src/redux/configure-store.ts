import { createStore, combineReducers, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise';
import {
  typePendingReducerSet,
  createTypeReduxInitialState,
  typeReduxMiddleware,
} from './type-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import * as AppReducer from './reducers/app-reducer';

export const rootReducer = combineReducers({
  ...typePendingReducerSet,
  appState: AppReducer.reducer,
});

export interface RootState {
  appState: AppReducer.State;
}

export const InitialState: RootState = Object.assign(createTypeReduxInitialState(), {
  appState: AppReducer.initialState,
});

const middlewares: any[] = [typeReduxMiddleware, promiseMiddleware];

export const configureStore = (initialState?: RootState) => {
  return createStore(
    rootReducer,
    { ...InitialState, ...initialState },
    composeWithDevTools(applyMiddleware(...middlewares)),
  );
};
