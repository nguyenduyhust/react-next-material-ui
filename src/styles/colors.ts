export const BLACK_COLOR = '#000000';
export const WHITE_COLOR = '#FFFFFF';
export const TORY_BLUE_COLOR = '#344090';
export const NAVY_BLUE_COLOR = '#1976D2';
export const PATTENS_BLUE_COLOR = '#D8DFE5';
export const MID_NIGHT_COLOR = '#233044';
export const ALICE_BLUE_COLOR = '#F7F9FC';
export const NOBEL_COLOR = '#9E9E9E';
export const CHARCOAL_COLOR = '#424242';
