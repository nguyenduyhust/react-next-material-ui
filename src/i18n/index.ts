import NextI18Next from 'next-i18next';
import path from 'path';
import moment from 'moment';
import { ns, defaultNS } from './ns';

const {
  Trans,
  Link,
  Router,
  i18n,
  initPromise,
  config,
  useTranslation,
  withTranslation,
  appWithTranslation,
} = new NextI18Next({
  ns,
  defaultNS,
  defaultLanguage: 'en',
  otherLanguages: ['ja'],
  browserLanguageDetection: true,
  serverLanguageDetection: true,
  detection: {
    order: ['querystring', 'cookie', 'header', 'navigator'],
    lookupQuerystring: 'lng',
    lookupCookie: 'lng',
    lookupHeader: 'accept-language',
    caches: ['cookie'],
  },
  localePath: path.resolve('public/static/locales'),
  interpolation: {
    format: function (value, format, lng) {
      if (value instanceof Date) {
        return moment(value).format(format);
      }
      return value;
    },
  },
});

export {
  Trans,
  Link,
  Router,
  i18n,
  initPromise,
  config,
  useTranslation,
  withTranslation,
  appWithTranslation,
};
export { NamespaceEnum } from './ns';
