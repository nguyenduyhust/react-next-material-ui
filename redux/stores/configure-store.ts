import { createStore, combineReducers, applyMiddleware } from 'redux';
import promiseMiddleware from 'redux-promise';
import { typePendingReducerSet, createTypeReduxInitialState, typeReduxMiddleware } from 'type-redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';

import * as AppReducer from './app-reducer';

export const rootReducer = combineReducers({
  ...typePendingReducerSet,
  appState: AppReducer.reducer,
});

export interface Store {
  appState: AppReducer.State;
}

export const InitialState: Store = Object.assign(createTypeReduxInitialState(), {
  appState: AppReducer.initialState,
});

const middlewares: any[] = [typeReduxMiddleware, promiseMiddleware];

export const configureStore = (initialState?: Store) => {
  return createStore(rootReducer, { ...InitialState, ...initialState }, composeWithDevTools(applyMiddleware(...middlewares)));
};
