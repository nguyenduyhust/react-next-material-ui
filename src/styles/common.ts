import { CSSProperties } from '@material-ui/styles';

export const link: CSSProperties = {
  textDecoration: 'none',
  '&:hover': {
    textDecoration: 'underline',
  },
};
