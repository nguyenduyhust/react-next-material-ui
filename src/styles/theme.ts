import { createMuiTheme, Theme } from '@material-ui/core/styles';
import {
  BLACK_COLOR,
  WHITE_COLOR,
  TORY_BLUE_COLOR,
  NAVY_BLUE_COLOR,
  PATTENS_BLUE_COLOR,
  MID_NIGHT_COLOR,
  ALICE_BLUE_COLOR,
  NOBEL_COLOR,
  CHARCOAL_COLOR,
} from '../constants/colors';
import {
  XS_BREAKPOINT,
  SM_BREAKPOINT,
  MD_BREAKPOINT,
  LG_BREAKPOINT,
  XL_BREAKPOINT,
} from '../constants/breakpoints';

const theme: Theme = createMuiTheme({
  colors: {
    black: BLACK_COLOR,
    white: WHITE_COLOR,
    toryBlue: TORY_BLUE_COLOR,
    navyBlue: NAVY_BLUE_COLOR,
    pattensBlue: PATTENS_BLUE_COLOR,
    midnight: MID_NIGHT_COLOR,
    aliceBlue: ALICE_BLUE_COLOR,
    nobel: NOBEL_COLOR,
    charcoal: CHARCOAL_COLOR,
  },
  breakpoints: {
    values: {
      xs: XS_BREAKPOINT,
      sm: SM_BREAKPOINT,
      md: MD_BREAKPOINT,
      lg: LG_BREAKPOINT,
      xl: XL_BREAKPOINT,
    },
  },
  spacing: 8,
  palette: {
    primary: {
      main: NAVY_BLUE_COLOR,
    },
  },
});

export default theme;
