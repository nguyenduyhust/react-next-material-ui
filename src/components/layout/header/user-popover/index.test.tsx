import React from 'react';
import { mount } from 'enzyme';
import firebase from 'firebase';
import { I18nextProvider } from 'react-i18next';
import { i18n } from '~/i18n';
import { SnackbarProvider } from 'notistack';
import { Provider } from 'react-redux';
import { configureStore } from '~/redux/configure-store';
import { ThemeProvider } from '@material-ui/core/styles';
import theme from '~/styles/theme';
import Component from '.';

describe('Component test', () => {
  test('Renders correctly', () => {
    const onClose = jest.fn();
    const user = {
      uid: 'uid',
      email: 'test@test.com',
      displayName: 'Test',
      photoURL: 'http://test.com',
    } as firebase.User;
    const anchorEl = document.createElement('button');
    const store = configureStore();
    const tree = mount(
      <SnackbarProvider>
        <I18nextProvider i18n={i18n}>
          <ThemeProvider theme={theme}>
            <Provider store={store}>
              <Component onClose={onClose} user={user} anchorEl={anchorEl} />
            </Provider>
          </ThemeProvider>
        </I18nextProvider>
      </SnackbarProvider>,
    ).html();
    expect(tree).toMatchSnapshot();
  });
});
