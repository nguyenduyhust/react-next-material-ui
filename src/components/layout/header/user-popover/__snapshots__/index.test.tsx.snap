// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component test Renders correctly 1`] = `"<div role=\\"presentation\\" class=\\"MuiPopover-root\\" style=\\"position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;\\"><div aria-hidden=\\"true\\" style=\\"z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: transparent;\\"></div><div tabindex=\\"0\\" data-test=\\"sentinelStart\\"></div><div class=\\"MuiPaper-root MuiPopover-paper MuiPaper-elevation8 MuiPaper-rounded\\" style=\\"opacity: 1; transform: scale(1, 1); transition: opacity 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 0ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; top: 16px; left: 16px; transform-origin: -16px -16px;\\" tabindex=\\"-1\\"><ul class=\\"MuiList-root MuiList-padding\\"><li class=\\"MuiListItem-root makeStyles-listItem-2 makeStyles-separateLine-3 MuiListItem-gutters\\"><img class=\\"makeStyles-avatar-4\\" src=\\"http://test.com\\" alt=\\"\\"><div><p class=\\"MuiTypography-root MuiTypography-body1\\">Test</p><span class=\\"MuiTypography-root makeStyles-email-5 MuiTypography-caption\\">test@test.com</span></div></li><div class=\\"MuiButtonBase-root MuiListItem-root makeStyles-listItem-2 makeStyles-separateLine-3 MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">setting<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M19.14 12.94c.04-.3.06-.61.06-.94 0-.32-.02-.64-.07-.94l2.03-1.58c.18-.14.23-.41.12-.61l-1.92-3.32c-.12-.22-.37-.29-.59-.22l-2.39.96c-.5-.38-1.03-.7-1.62-.94l-.36-2.54c-.04-.24-.24-.41-.48-.41h-3.84c-.24 0-.43.17-.47.41l-.36 2.54c-.59.24-1.13.57-1.62.94l-2.39-.96c-.22-.08-.47 0-.59.22L2.74 8.87c-.12.21-.08.47.12.61l2.03 1.58c-.05.3-.09.63-.09.94s.02.64.07.94l-2.03 1.58c-.18.14-.23.41-.12.61l1.92 3.32c.12.22.37.29.59.22l2.39-.96c.5.38 1.03.7 1.62.94l.36 2.54c.05.24.24.41.48.41h3.84c.24 0 .44-.17.47-.41l.36-2.54c.59-.24 1.13-.56 1.62-.94l2.39.96c.22.08.47 0 .59-.22l1.92-3.32c.12-.22.07-.47-.12-.61l-2.01-1.58zM12 15.6c-1.98 0-3.6-1.62-3.6-3.6s1.62-3.6 3.6-3.6 3.6 1.62 3.6 3.6-1.62 3.6-3.6 3.6z\\"></path></svg><span class=\\"MuiTouchRipple-root\\"></span></div><div class=\\"MuiButtonBase-root MuiListItem-root makeStyles-listItem-2 MuiListItem-gutters MuiListItem-button\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\">header.sign_out<svg class=\\"MuiSvgIcon-root\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10.09 15.59L11.5 17l5-5-5-5-1.41 1.41L12.67 11H3v2h9.67l-2.58 2.59zM19 3H5c-1.11 0-2 .9-2 2v4h2V5h14v14H5v-4H3v4c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z\\"></path></svg><span class=\\"MuiTouchRipple-root\\"></span></div></ul></div><div tabindex=\\"0\\" data-test=\\"sentinelEnd\\"></div></div>"`;
